apis/tags/v1beta1/tagbinding_parentref.go

```diff
@@ -62,39 +69,51 @@ func (r *TagsTagBindingParentRef) SetExternal(ref string) {
 }
 
 func (r *TagsTagBindingParentRef) ValidateExternal(ref string) error {
-	gvk := r.GetGVK()
-	switch gvk {
-	case refs.ProjectGVK:
-		id := &refs.ProjectRef{
-			Name:      r.Name,
-			Namespace: r.Namespace,
-			External:  r.External,
-		}
-		if err := id.ValidateExternal(ref); err != nil {
-			return err
-		}
-		return nil
-	default:
-		return fmt.Errorf("format of %s external=%q was not known", gvk.Kind, ref)
+	_, id := r.resolveReference()
+	if id == nil {
+		return fmt.Errorf("unknown format for external reference %q", r.External)
 	}
+
+	if err := id.ValidateExternal(ref); err != nil {
+		return err
+	}
+	return nil
 }
 
 func (r *TagsTagBindingParentRef) Normalize(ctx context.Context, reader client.Reader, defaultNamespace string) error {
-	gvk := r.GetGVK()
-	switch gvk {
-	case refs.ProjectGVK:
-		id := &refs.ProjectRef{
-			Name:      r.Name,
-			Namespace: r.Namespace,
-			External:  r.External,
-		}
-		if err := id.Normalize(ctx, reader, defaultNamespace); err != nil {
-			return err
+	service, id := r.resolveReference()
+	if id == nil {
```

Comment:
This `if/else if` chain for determining the reference type is hard to maintain. A `map[string]refs.Ref` or a similar data structure could make this logic more data-driven and easier to extend with new types in the future.